src/AbuseIpDb.php

```diff
@@ -54,6 +70,30 @@ public function IsSpamIp()
         return false;
     }
 
+    /**
+     * @param string $categories The comma separated categories to report (eg 18,22), see https://www.abuseipdb.com/categories
+     * @param string $ip (optional) The ip address to report. Or set it before with $this->setIp($ip)
+     * @param string $comment Related information (server logs, timestamps, etc.)
+     * @return bool The updated abuse confident score
+     * @throws AbuseIpDbException
+     */
+    public function reportIp($categories, $ip = null, $comment = '')
```

Comment:
What is the reason to use/implement this (the ip) compared with the current setIp? Better (work)-flow?

Reply:
yes, but just my pov.